clients/coredata/event.go

```diff
@@ -68,37 +67,24 @@ type EventClient interface {
 }
 
 type eventRestClient struct {
-	url      string
-	endpoint interfaces.Endpointer
+	urlClient interfaces.URLClient
 }
 
 // NewEventClient creates an instance of EventClient
 func NewEventClient(params types.EndpointParams, m interfaces.Endpointer) EventClient {
-	e := eventRestClient{endpoint: m}
-	e.init(params)
-	return &e
-}
-
-func (e *eventRestClient) init(params types.EndpointParams) {
-	if params.UseRegistry {
-		go func(ch chan string) {
-			for {
-				select {
-				case url := <-ch:
-					e.url = url
-				}
-			}
-		}(e.endpoint.Monitor(params))
-	} else {
-		e.url = params.Url
-	}
+	return &eventRestClient{urlClient: urlclient.New(params, m)}
 }
 
 // Helper method to request and decode an event slice
-func (e *eventRestClient) requestEventSlice(url string, ctx context.Context) ([]models.Event, error) {
-	data, err := clients.GetRequest(url, ctx)
+func (e *eventRestClient) requestEventSlice(urlSuffix string, ctx context.Context) ([]models.Event, error) {
+	urlPrefix, err := e.urlClient.Prefix()
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := clients.GetRequest(urlPrefix+urlSuffix, ctx)
 	if err != nil {
-		return []models.Event{}, err
+		return nil, err
```

Comment:
Not sure if this should be nil or an empty slice of models.Event.

Golang idiom is to return nil.  EdgeX idiom is to return empty slice of models.Event.  We need to be consistent with EdgeX idiom.

See comments immediately above and below.

May occur elsewhere.